server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java

```diff
@@ -8,180 +8,53 @@
-    }
-
-    @Override
-    public Query wildcardQuery(String value, MultiTermQuery.RewriteMethod method, boolean caseInsensitive, SearchExecutionContext context) {
-        throw new IllegalArgumentException(unsupported("wildcard", "keyword, text and wildcard"));
-    }
+    void doXContentBody(XContentBuilder builder, Params params) throws IOException;
 
-    @Override
-    public Query regexpQuery(
-        String value,
-        int syntaxFlags,
-        int matchFlags,
-        int maxDeterminizedStates,
-        MultiTermQuery.RewriteMethod method,
-        SearchExecutionContext context
-    ) {
-        throw new IllegalArgumentException(unsupported("regexp", "keyword and text"));
-    }
-
-    @Override
-    public Query phraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements) {
-        throw new IllegalArgumentException(unsupported("phrase", "text"));
-    }
-
-    @Override
-    public Query multiPhraseQuery(TokenStream stream, int slop, boolean enablePositionIncrements) {
-        throw new IllegalArgumentException(unsupported("phrase", "text"));
-    }
-
-    @Override
-    public Query phrasePrefixQuery(TokenStream stream, int slop, int maxExpansions) {
-        throw new IllegalArgumentException(unsupported("phrase prefix", "text"));
-    }
-
-    @Override
-    public SpanQuery spanPrefixQuery(String value, SpanMultiTermQueryWrapper.SpanRewriteMethod method, SearchExecutionContext context) {
-        throw new IllegalArgumentException(unsupported("span prefix", "text"));
-    }
-
-    private String unsupported(String query, String supported) {
-        return String.format(
-            Locale.ROOT,
-            "Can only use %s queries on %s fields - not on [%s] which is a runtime field of type [%s]",
-            query,
-            supported,
-            name(),
-            typeName()
-        );
-    }
+    /**
+     * Exposes the name of the runtime field
+     * @return name of the field
+     */
+    String name();
 
-    protected final void checkAllowExpensiveQueries(SearchExecutionContext context) {
-        if (context.allowExpensiveQueries() == false) {
-            throw new ElasticsearchException(
-                "queries cannot be executed against runtime fields while [" + ALLOW_EXPENSIVE_QUERIES.getKey() + "] is set to [false]."
-            );
-        }
-    }
+    /**
+     * Exposes the type of the runtime field
+     * @return type of the field
+     */
+    String typeName();
 
-    @Override
-    public final ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
-        return new DocValueFetcher(docValueFormat(format, null), context.getForField(this));
-    }
+    /**
+     * Exposes the {@link MappedFieldType} backing this runtime field, used to execute queries, run aggs etc.
+     * @return the {@link MappedFieldType} backing this runtime field
+     */
+    MappedFieldType asMappedFieldType();
```

Comment:
I envision this returning a Collection<MappedFieldType> once we introduce the object runtime field